packages/gestalt/src/Layer.js

```diff
@@ -1,54 +1,36 @@
 // @flow strict
-import { Component, type Portal, type Node } from 'react';
+import { useRef, useState, type Portal, type Node, useEffect } from 'react';
 // flowlint-next-line untyped-import:off
 import { createPortal } from 'react-dom';
 
-type Props = {|
+export default function Layer({
+  children,
+}: {|
   children: Node,
-|};
+|}): Portal | null {
+  const [mounted, setMounted] = useState(false);
+  const element = useRef<?HTMLDivElement>(null);
 
-type State = {|
-  mounted: boolean,
-|};
+  useEffect(() => {
+    setMounted(true);
 
-export default class Layer extends Component<Props, State> {
-  state: State = {
-    mounted: false,
-  };
-
-  el: HTMLDivElement;
-
-  constructor(props: Props) {
-    super(props);
     if (typeof document !== 'undefined' && document.createElement) {
-      this.el = document.createElement('div');
-    } else {
-      // eslint-disable-next-line no-console
-      console.warn(
-        'Using Layer without document present. Children will not be rendered.'
```

Comment:
The idea for this was to provide a warning if you tried the use `Layer` during SSR since that wouldn't work. Do you think that isn't useful?

Reply:
@jennyscript this is the only place where we do this and I think it would be better for it not to become a pattern. Ideally we would solve this through linting instead of adding `console.warn` / `console.log`